src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,84 @@
+import { useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../utils/PersonLink';
+import classNames from 'classnames';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slugId } = useParams();
+  const findPersonByName = (name: string) => {
+    return people.find(person => person.name === name);
+  };
+
+  const peopleWithParents = people.map(person => {
+    const mother = person.motherName
+      ? findPersonByName(person.motherName)
+      : undefined;
+    const father = person.fatherName
+      ? findPersonByName(person.fatherName)
+      : undefined;
+
+    return { ...person, mother, father };
+  });
```

Comment:
Code quality issue: This block of code is doing a lot of work and making the component quite complex. Consider moving this data processing outside of the component, perhaps in a useMemo hook or even outside of this file entirely. This would make the component code cleaner and easier to understand.